pipeline/authn/authenticator_oauth2_introspection.go

```diff
@@ -204,6 +204,10 @@ func (a *AuthenticatorOAuth2Introspection) Authenticate(r *http.Request, session
 		i.Extra["client_id"] = i.ClientID
 		i.Extra["scope"] = i.Scope
 
+		if len(i.Audience) != 0 {
+			i.Extra["aud"] = strings.Join(i.Audience, " ")
```

Comment:
Audience is a string of one element or an array of multiple but not a string concatenation

```suggestion
			i.Extra["aud"] = i.Audience
```